app/code/Bulbulatory/Recomendations/Model/RecommendationRepository.php

```diff
@@ -0,0 +1,113 @@
+<?php
+ 
+namespace Bulbulatory\Recomendations\Model;
+ 
+use Magento\Framework\Api\SearchCriteriaInterface;
+use Magento\Framework\Api\SortOrder;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Bulbulatory\Recomendations\Api\Data\RecommendationInterface;
+use Bulbulatory\Recomendations\Api\Data\RecommendationSearchResultInterface;
+use Bulbulatory\Recomendations\Api\Data\RecommendationSearchResultInterfaceFactory;
+use Bulbulatory\Recomendations\Api\RecommendationRepositoryInterface;
+use Bulbulatory\Recomendations\Model\ResourceModel\Recommendation\CollectionFactory as RecommendationCollectionFactory;
+use Bulbulatory\Recomendations\Model\ResourceModel\Recommendation\Collection;
+ 
+class RecommendationRepository implements RecommendationRepositoryInterface
+{
+    /**
+     * @var RecommendationFactory
+     */
+    private $recommendationFactory;
+ 
+    /**
+     * @var RecommendationCollectionFactory
+     */
+    private $recommendationCollectionFactory;
+ 
+    /**
+     * @var RecommendationSearchResultInterfaceFactory
+     */
+    private $searchResultFactory;
+ 
+    public function __construct(
+        RecommendationFactory $recommendationFactory,
+        RecommendationCollectionFactory $recommendationCollectionFactory,
+        RecommendationSearchResultInterfaceFactory $recommendationSearchResultInterfaceFactory
+    ) {
+        $this->recommendationFactory = $recommendationFactory;
+        $this->recommendationCollectionFactory = $recommendationCollectionFactory;
+        $this->searchResultFactory = $recommendationSearchResultInterfaceFactory;
+    }
+
+    public function getById($id)
+    {
+        $recommendation = $this->recommendationFactory->create();
+        $recommendation->getResource()->load($recommendation, $id);
```

Comment:
Składnia `$model->getResource()` jest oznaczona jako deprecated, powinno się używać resource bezpośrednio, czyli przekazać w konstruktorze obiekt klasy `\Bulbulatory\Recomendations\Model\ResourceModel\Recommendation` i jego używać:
```suggestion
        $$this->recommendationResource>load($recommendation, $id);
```
To samo dotyczy pozostałych metod z tej klasy.